### DISCUSSION TITLE
ALTER TAG or ALTER EDGE - Is such change reflected immediately?

### DISCUSSION THREAD

**Discussion by xs2tarunkukreja** [vesoft-inc/nebula#5658] [Q&A] [answered]
We have a requirement to add new property in tags as well as edges. We have millions of vertex and edges. 

Will changing the tag or vertex be effected immediately? 

If not then how we can know that it is effected now.
@wey-gu

**Comment by wey-gu** [comment]
It's done on metad when it succeeded, but the underlying data will be changed during further data updates and (storage periodic)compaction.

But the effect was taken as a query after a successful TAG/EDGE ALTER AND two Heartbeat cycles, to distribute the schema change to all services) will have resulted in the new SCHEMA.

**Reply by xs2tarunkukreja** [reply]
@wey-gu 
What is time for 2 heartbeat cycle?

**Reply by xs2tarunkukreja** [reply]
I have read somewhere it is 20 seconds. Is it static? If not then how can we know the value of the same?

**Reply by wey-gu** [reply, marked as answer]
> I have read somewhere it is 20 seconds. Is it static? If not then how can we know the value of the same?
> 
> 

20 sec is generally true.

Default heartbeat is indeed 10 sec, it's configurable though.
